Adds JS/TS toggle to interactive docs samples

aomarks opened the pull request:
Adds a JS/TS toggle to all interactive samples in the docs, behind the `?mods=jsSamples` feature flag.

**Note:** The playground page, tutorial page, and static samples will come in a followup PR.

**Also note:** There are a some issues with the JS samples that I've seen already, mostly around `/*playground-fold*/` and `/*playground-hide*/` comments not getting preserved quite right. I'll address those in a followup too.

Also fixes layout shift issues with inline playgrounds.

#### With tabs
<img src="https://user-images.githubusercontent.com/48894/132420551-6f700813-71b6-4b58-9cc0-80c6c891e0f3.png" width="500px">

#### Without tabs
<img src="https://user-images.githubusercontent.com/48894/132420560-89c60b17-bf9b-482e-ae0d-c3b468e7dec1.png" width="500px">

AndrewJakubowicz reviewed (commented):
Before looking at code, some quick nits from manual testing.

See screenshot comparing https://lit.dev/docs/components/properties/ and https://pr478-ff9f2da---lit-dev-5ftespv5na-uc.a.run.app/docs/components/properties/

Screenshot is of the last interactive example on the page using Chrome.

![Screen Shot 2021-09-07 at 4 26 48 PM](https://user-images.githubusercontent.com/15080861/132422892-9817001b-76ae-4767-98c7-ee39664dbb5a.png)

 - `<div id="bar">` is possibly missing border-radius. Visual artifact is the rounded corner has a tiny break in it.
 - Last example has a scroll bar. This is fine, but maybe could add some padding beneath the "open example in playground" and the top of the scroll bar.

AndrewJakubowicz commented on packages/lit-dev-content/site/css/mods.css:
```diff
@@ -22,3 +22,9 @@
+.jsSamples {
+  /* TODO(aomarks) Remove when JS samples launch, and update the
+  litdev-typescript-switch display property to match this. */
+  --litdev-typescript-switch-display: inline-flex;
```
Genius. The mod system is brilliant.

AndrewJakubowicz commented on packages/lit-dev-content/src/components/litdev-example-controls.ts:
```diff
@@ -0,0 +1,66 @@
+ * Controls for lit.dev code examples.
+ */
+@customElement('litdev-example-controls')
+export class LitDevExampleControls extends LitElement {
```
Minor style nitpick. The switch and openInPlayground link have different heights, 25px and 22px respectively.

This is non blocking since the switch isn't even enabled. Instead commenting that it currently feels like the switch is sitting lower in the menu bar than the open in playground image. The two menu items don't feel vertically aligned.

aomarks replied:
Here are some options:

Before:
<img src="https://user-images.githubusercontent.com/48894/132425290-cef94bf7-4752-4642-9a1e-70328d34efdd.png" width="150px">

Top-aligned:
<img src="https://user-images.githubusercontent.com/48894/132425285-2252644a-2e1d-4da0-9ea7-f58b358138a1.png" width="150px">

Enlarged:
<img src="https://user-images.githubusercontent.com/48894/132424872-8fcb7582-6147-4eff-ae63-a22e3dae3a53.png" width="150px">

Is "top-aligned" what you were thinking? Or more like "enlarged"? The problem with "enlarged" to my eye is that the stroke width of the icon becomes too heavy, and that clashes with the stroke width of the JS/TS switch. We could edit the SVG to drop the stroke width, though. WDYT?

AndrewJakubowicz replied:
Let's keep it as before. I think my eye is getting impacted by the arrow at the top of the icon making it feel like it's higher up.

I was thinking more along the lines of enlarged but 100% agree about the stroke width making it awkward.

justinfagnani replied:
I like "before" of these... my understanding of the material sizing system (likely imperfect) is that there's an outer size, like 24px, but the icon doesn't fully fill it, which some components, like this switch might. So the container's hsould have the same size, but the icon-based widgets will appear a bit smaller.

aomarks replied:
Done!

aomarks commented:
> * `<div id="bar">` is possibly missing border-radius. Visual artifact is the rounded corner has a tiny break in it.

Good catch, that was it! Fixed.

> * Last example has a scroll bar. This is fine, but maybe could add some padding beneath the "open example in playground" and the top of the scroll bar.

Good catch. I think actually a lot of the samples are a weird size right now, even for TypeScript. I'll do a sweep through all of them as a followup to improve the sizes. (I've been planning to automatically compute the sizes during build, but still haven't gotten around to it, so right now they are all defined manually and they get out of sync easily as the code changes.)

> * The `JS` side of the toggle is `~40px` while the `TS` side is `~45px`. This is a very micro nit, but it feels like it should be equal size.

Oh right, it was using the intrinsic size of the content, so it was very slightly different. Fixed so they are equal size.

AndrewJakubowicz commented on packages/lit-dev-content/src/components/litdev-typescript-switch.ts:
```diff
@@ -0,0 +1,127 @@
+  `;
+
+  override connectedCallback() {
+    // TODO(aomarks) After we upgrade to Lit 2, this and similar code in other
+    // components could be refactored into a controller.
```
Exceptional comment. Was wondering this.

AndrewJakubowicz reviewed (approved):
Brilliant! LGTM!

AndrewJakubowicz commented on packages/lit-dev-content/src/components/litdev-example-controls.ts:
```diff
@@ -49,7 +46,7 @@ export class LitDevExampleControls extends LitElement {
       >
         <!-- Source: https://material.io/resources/icons/?icon=launch&style=baseline -->
-        <svg width="22px" height="22px" viewBox="0 0 24 24" fill="#5f5f5f">
+        <svg width="22px" height="22px" viewBox="0 2 24 24" fill="#5f5f5f">
```
I really like this 2px bump down.

AndrewJakubowicz commented on packages/lit-dev-content/src/components/litdev-example-controls.ts:
```diff
@@ -15,9 +15,6 @@ export class LitDevExampleControls extends LitElement {
       display: flex;
-      flex-direction: row;
-      align-items: center;
-      justify-content: center;
```
I personally prefer this with the centering especially when there are file tabs.

![Screen Shot 2021-09-07 at 5 22 40 PM](https://user-images.githubusercontent.com/15080861/132426144-8b2eb5f2-a182-4ec5-93c2-f01743309310.png)

aomarks replied:
Oh yeah, definitely looks wrong here :D

justinfagnani commented:
Is there a "switch" component that could be used? I think that would be more intuitive/obvious.

This switch doesn't look that switchable at first glance to me, in that the options are mutually exclusive. Maybe some styling tweaks could help - like making the non-selected option more grayed out, and/or share a background color with its container?

aomarks commented:
> Is there a "switch" component that could be used? I think that would be more intuitive/obvious.
> 
> This switch doesn't look that switchable at first glance to me, in that the options are mutually exclusive. Maybe some styling tweaks could help - like making the non-selected option more grayed out, and/or share a background color with its container?

Yeah, I could see that. The current design is what I think a Material-ish to this problem would look like. A totally different option (no reason we have to stick with Material) is something like this:

<img src="https://user-images.githubusercontent.com/48894/132427827-9046620f-25e8-424b-89f8-d385ea95b5a9.png" width="200px">

I'm going to punt on this for now, since the switch feature is behind a feature flag, but I'll make sure to revisit and potentially implement something like above before the public launch.
